File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if(n==1) return 1;
        int p1=1, p2=2;
        int sum;
        for(int i=3;i<=n;i++){
            sum=p1+p2;
            p1=p2;
            p2=sum;
        }   
        return p2;
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n<2){
            return n;
        }
        int p1=0;
        int p2=1;
        // int res=0;
        for(int i=1;i<=n;i++){
            int temp=p1+p2;
            p1=p2;
            p2=temp;
        }
        return p2;
    }
}